modules/replicator/src/main/java/org/apache/ignite/internal/replicator/ReplicaManager.java

```diff
@@ -926,6 +959,23 @@ public CompletableFuture<Void> stopAsync(ComponentContext componentContext) {
         shutdownAndAwaitTermination(executor, shutdownTimeoutSeconds, TimeUnit.SECONDS);
         shutdownAndAwaitTermination(replicasCreationExecutor, shutdownTimeoutSeconds, TimeUnit.SECONDS);
 
+        try {
+            CompletableFuture.allOf(raftClientsFutures.values().toArray(new CompletableFuture<?>[0]))
+                    .get(shutdownTimeoutSeconds, TimeUnit.SECONDS);
+
+            raftClientsFutures.forEach((raftNodeId, raftClientFuture) -> {
+                raftClientFuture.thenAccept(RaftGroupService::shutdown);
+
+                try {
+                    raftManager.stopRaftNode(raftNodeId);
+                } catch (NodeStoppingException e) {
```

Comment:
I don't think that it's possible to catch following exception here. Could you please elaborate a scenario?

Reply:
Fixed with a proper `AssertionError` with a message why it's an abnormal case